google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1beta2/JsonToProtoMessage.java

```diff
@@ -234,7 +238,21 @@ private static void fillRepeatedField(
           break;
         case BYTES:
           if (val instanceof String) {
+            // TODO(jstocklass): If string, decode it and pass in the byte array. Will need to
+            // update
+            // tests to ensure that strings passed in are properly encoded as well.
             protoMsg.addRepeatedField(fieldDescriptor, ((String) val).getBytes());
+          } else if (val instanceof JSONArray) {
+            try {
+              byte[] bytes = new byte[((JSONArray) val).length()];
+              for (int j = 0; j < ((JSONArray) val).length(); j++) {
+                bytes[j] = (byte) ((byte) (((JSONArray) val).get(j)) & 0xFF);
+                // TODO(jstocklass): Add a unit test for this
```

Comment:
Add negative tests.

Reply:
I've added a negative test as well now, it covers the failure if the json cannot be converted to bytes and we get the new error message.